internal/pkg/fetch/fetch.go

```diff
@@ -101,14 +97,18 @@ func fetchExternalReferences(document *sbom.Document, backend *db.Backend, opts
 	}
 
 	for _, ref := range extRefs {
-		refOutput, err := getRefFile(opts.OutputFile)
-		if err != nil {
-			return err
+		extRefsOpt := *opts
```

Comment:
Is this actually doing a copy by value, or are the two variables still referencing the same underlying memory address?

What does it say if, for debugging purposes, you add this before and after the call to `getRefFile()`:

```go
fmt.Printf("opts: %p\n", opts.OutputFile)
fmt.Printf("extRefsOpt: %p\n", extRefsOpt.OutputFile)
```